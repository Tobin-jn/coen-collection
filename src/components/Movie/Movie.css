.Movie {
  background-color: black;
  border-radius: 40px;
  height: 390px;
  margin: 16px;
  width: 260px;
}

.movie-details,
.movie-poster {
  border-radius: 30px;
  cursor: pointer;
  height: 390;
  opacity: 0.9;
  width: 260px;
}

.movie-details {
  background-image: linear-gradient(silver, white);
  height: 390px;
  line-height: 1.2;
  text-align: center;
}

.movie-details:hover,
.movie-poster:hover {
  bottom: 10px;
  display: block;
  height: 400px;
  opacity: 1;
  position: relative;
  right: 10px;
  width: 280px;
}

.movie-details:hover {
  opacity: 0.9;
  overflow: scroll;
}

.movie-date {
  margin: 10px;
}

.text-wrapper {
  margin: auto;
  padding: 20px 0;
  width: 230px;
}
