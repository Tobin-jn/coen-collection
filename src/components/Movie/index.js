import React from 'react';
import './Movie.css'

const Movie = () => {
  return (
    <div className="Movie">Movie</div>
  )
}

export default Movie