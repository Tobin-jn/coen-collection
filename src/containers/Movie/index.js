import React, { Component } from "react";
import PropTypes from "prop-types";
import "./Movie.css";
import { connect } from "react-redux";
import { toggleFavorite, setErrorMessage } from "../../actions";
import * as fetch from "../../utilities/fetch.js";

export class Movie extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isHovered: false
    };
  }

  handleClickFavorite = () => {
    const {
      favorited,
      id,
      handleToggleFavorite,
      handleErrorMessage
    } = this.props;
    if (!this.verifyUserIsLoggedIn()) {
      handleErrorMessage("Please log in or sign up to select favorites");
      return;
    }

    handleToggleFavorite(id);
    if (!favorited) {
      try {
        fetch.fetchAddFavorite(this.props);
      } catch (error) {}
    } else {
      fetch.removeFavorite(this.props.currentUser.id, id);
    }
  };

  handleHover = bool => {
    const isHovered = bool;
    this.setState({ isHovered });
  };

  verifyUserIsLoggedIn = () => {
    return this.props.loggedIn;
  };

  render() {
    let { title, overview, date, poster, favorited } = this.props;
    let cardContents;

    let year = "/" + date.split("-")[0];
    date = date
      .split("-")
      .slice(1)
      .join("/");

    if (!this.state.isHovered) {
      cardContents = (
        <div>
          <i className={"fa-star" + (favorited ? " fas favorited" : " far")} />
          <img className="movie-poster" src={poster} alt="Movie Poster" />
        </div>
      );
    } else {
      cardContents = (
        <div className="movie-details">
          <article className="text-wrapper">
            <i
              className={"fa-star" + (favorited ? " fas favorited" : " far")}
              onClick={this.handleClickFavorite}
            />
            <h1 className="movie-title">{title}</h1>
            <p className="movie-date">
              {date}
              {year}
            </p>
            <p className="movie-overview">{overview}</p>
          </article>
        </div>
      );
    }
    return (
      <div
        onMouseLeave={() => this.handleHover(false)}
        onMouseEnter={() => this.handleHover(true)}
        className="Movie"
      >
        {cardContents}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  currentUser: state.currentUser,
  loggedIn: state.loggedIn
});

export const mapDispatchToProps = dispatch => ({
  handleToggleFavorite: id => dispatch(toggleFavorite(id)),
  handleErrorMessage: message => dispatch(setErrorMessage(message))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Movie);

Movie.propTypes = {
  title: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  date: PropTypes.number.isRequired,
  voteAverage: PropTypes.number.isRequired,
  poster: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  favorited: PropTypes.bool.isRequired,
  currentUser: PropTypes.object.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  handleToggleFavorite: PropTypes.func.isRequired,
  handleErrorMessage: PropTypes.func.isRequired
};
