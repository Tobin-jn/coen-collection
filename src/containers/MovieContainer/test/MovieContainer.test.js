/* eslint-disable */

import React from "react";
import { shallow } from "enzyme";
import { mapStateToProps, MovieContainer } from "../index";
import Movie from "../../../containers/Movie";

describe("MovieContainer", () => {
  let wrapper;
  let mockMovies

  beforeEach(() => {
    mockMovies = [
      {
        title: "The Big Lebowski",
        date: "1998-03-06",
        poster:
          "https://image.tmdb.org/t/p/w600_and_h900_bestv2/aHaVjVoXeNanfwUwQ92SG7tosFM.jpg",
        overview:
          'Jeffrey "The Dude" Lebowski, a Los Angeles slacker who only wants to bowl and drink white Russians, is mistaken for another Jeffrey Lebowski, a wheelchair-bound millionaire, and finds himself dragged into a strange series of events involving nihilists, adult film producers, ferrets, errant toes, and large sums of money.',
        voteAverage: 7.9,
        id: 115,
        favorited: false
      }
    ];

    wrapper = shallow(<MovieContainer 
      movies={mockMovies}
      showFavorites={true}
      userName={'Taylor'} 
    />);
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render like snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render cards if there are movies in the movies array", () => {

    expect(wrapper.find(Movie)).toBeDefined();
  });

  it("should render the spinner image if there are no movies in the movies array", () => {
    mockMovies = []

    wrapper = shallow(<MovieContainer 
      movies={mockMovies}
      showFavorites={true}
      userName={'Taylor'} 
    />);

    expect(wrapper.find(".spinner")).toBeDefined();
  });
});

describe("mapStateToProps", () => {
  it("should return an object with a movies array", () => {
    
    const mockState = {
      movies: [{ title: "Raising Arizona" }],
      showFavorites: false,
      currentUser: { name: 'Taylor', id: 4} 
    };

    const expected = {
      movies: [{ title: "Raising Arizona" }],
      showFavorites: false,
      userName: 'Taylor'

    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
  // it("should return an object with showFavorites", () => {
  //   const mockState = {
  //   };

  //   const expected = {
  //   };

  //   const mappedProps = mapStateToProps(mockState);
  //   expect(mappedProps).toEqual(expected);
  // });
  // it("should return an object with a userName", () => {
  //   const mockState = {
  //   };

  //   const expected = {
  //   };

  //   const mappedProps = mapStateToProps(mockState);
  //   expect(mappedProps).toEqual(expected);
  // });
});
