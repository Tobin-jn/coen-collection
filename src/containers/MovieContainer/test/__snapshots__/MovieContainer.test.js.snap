// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieContainer
    movies={
      Array [
        Object {
          "date": "1998-03-06",
          "favorited": false,
          "id": 115,
          "overview": "Jeffrey \\"The Dude\\" Lebowski, a Los Angeles slacker who only wants to bowl and drink white Russians, is mistaken for another Jeffrey Lebowski, a wheelchair-bound millionaire, and finds himself dragged into a strange series of events involving nihilists, adult film producers, ferrets, errant toes, and large sums of money.",
          "poster": "https://image.tmdb.org/t/p/w600_and_h900_bestv2/aHaVjVoXeNanfwUwQ92SG7tosFM.jpg",
          "title": "The Big Lebowski",
          "voteAverage": 7.9,
        },
      ]
    }
    showFavorites={true}
    userName="Taylor"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Walter Sobchak"
          className="walter"
          src="walter.jpg"
        />,
        <h3>
          You're entering a world of pain, Taylor. Pick some favorites!
        </h3>,
      ],
      "className": "Movie-Container no-favs",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Walter Sobchak",
          "className": "walter",
          "src": "walter.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You're entering a world of pain, Taylor. Pick some favorites!",
        },
        "ref": null,
        "rendered": "You're entering a world of pain, Taylor. Pick some favorites!",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Walter Sobchak"
            className="walter"
            src="walter.jpg"
          />,
          <h3>
            You're entering a world of pain, Taylor. Pick some favorites!
          </h3>,
        ],
        "className": "Movie-Container no-favs",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Walter Sobchak",
            "className": "walter",
            "src": "walter.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You're entering a world of pain, Taylor. Pick some favorites!",
          },
          "ref": null,
          "rendered": "You're entering a world of pain, Taylor. Pick some favorites!",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
